swift-sdk/IterableAuthManagerProtocol.swift

```diff
@@ -7,7 +7,11 @@ import Foundation
 @objc public protocol IterableAuthManagerProtocol {
     func getAuthToken() -> String?
     func resetFailedAuthCount()
-    func requestNewAuthToken(hasFailedPriorAuth: Bool, onSuccess: ((String?) -> Void)?)
+    func requestNewAuthToken(hasFailedPriorAuth: Bool, onSuccess: ((String?) -> Void)?, shouldIgnoreRetryPolicy: Bool)
+    func scheduleAuthTokenRefreshTimer(interval: TimeInterval, isScheduledRefresh: Bool, successCallback: AuthTokenRetrievalHandler?)
     func setNewToken(_ newToken: String)
     func logoutUser()
+    func pauseAuthRetries(_ pauseAuthRetry: Bool)
+    func setIsLastAuthTokenValid(_ isValid: Bool)
+    func getNextRetryInterval() -> Double
```

Comment:
Why is in - `JWT-InSettings` - branch
`   func getRetryCount() -> Int `
  `  func getPauseAuthRetry() -> Bool`
were added? 

It looks like adding these two functionalities will deviate the SDK to have different source of truth for total retry count and getPauseAuthRetry. 
We should limit the protocol methods with whats here in this PR

Reply:
Yes @Ayyanchira actually to get the JWT demo working properly (with correct values of retry count and to pause it using authmanager) we had to expose authManager along with these additional methods.